advanced/Templates/pihole-FTL.service

```
@@ -33,10 +38,15 @@ start() {
     sh "${PI_HOLE_SCRIPT_DIR}/pihole-FTL-prestart.sh"
 
     if setcap CAP_NET_BIND_SERVICE,CAP_NET_RAW,CAP_NET_ADMIN,CAP_SYS_NICE,CAP_IPC_LOCK,CAP_CHOWN+eip "/usr/bin/pihole-FTL"; then
-      su -s /bin/sh -c "/usr/bin/pihole-FTL" pihole || exit $?
+      su -s /bin/sh -c "/usr/bin/pihole-FTL" pihole || ec=$?
     else
       echo "Warning: Starting pihole-FTL as root because setting capabilities is not supported on this system"
-      /usr/bin/pihole-FTL || exit $?
+      /usr/bin/pihole-FTL || ec=$?
+    fi
+    # Cleanup if startup failed
+    if [ -n "${ec}" ] && [ "${ec}" != 0 ]; then
```

```suggestion
    if [ -n "${ec}" ]; then
```
`ec` is never `0` as it is applied only if FTL failed (`||`). But the extra check does not hurt either, of course.

Or to have it better understandable and failsafe against exported/passed `ec` env var:
```sh
    if setcap CAP_NET_BIND_SERVICE,CAP_NET_RAW,CAP_NET_ADMIN,CAP_SYS_NICE,CAP_IPC_LOCK,CAP_CHOWN+eip "/usr/bin/pihole-FTL"; then
      su -s /bin/sh -c "/usr/bin/pihole-FTL" pihole
    else
      echo "Warning: Starting pihole-FTL as root because setting capabilities is not supported on this system"
      /usr/bin/pihole-FTL
    fi
    rc=$?
    # Cleanup if startup failed
    if [ "${rc}" != 0 ]; then
        cleanup
        exit $ec
    fi

---

Thanks. I added your suggestions.